cmake/everest-generate.cmake

```diff
@@ -127,171 +133,229 @@ function (ev_add_modules)
     )
 endfunction()
 
+function(ev_setup_cpp_module)
+    # no-op to not break API
+endfunction()
 
-#
-# ev_setup_cpp_module, ev_setup_js_module and ev_setup_js_module are macros not functions for a deliberate reason:
-# calling return() in them allows us to stop processing the calling CMakeLists.txt
-#
-
-macro(ev_setup_cpp_module)
-    get_filename_component(MODULE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+function (ev_add_cpp_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
 
-    if(${EVEREST_EXCLUDE_CPP_MODULES})
-        message(STATUS "Excluding C++ module ${MODULE_NAME} because EVEREST_EXCLUDE_CPP_MODULES=${EVEREST_EXCLUDE_CPP_MODULES}")
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
```

Comment:
these two `set` statements seem to appear in all three add_* functions.  This is a good spot for using a private macro like `macro (_ev_module_common_setup)` or so, which sets these variables, otherwise one would need to change all 3 of them, if there will be changes needed